database/db.go

```diff
@@ -763,6 +763,29 @@ func (db *DB) Delete(
 	return db.DeleteStreamed(ctx, entityType, idsCh, onSuccess...)
 }
 
+// RunInTx allows running a function in a database transaction without requiring manual transaction handling.
+//
+// A new transaction is started on [DB] which is then passed to fn. After fn returns, the transaction is
+// committed unless an error was returned. If fn returns an error, that error is returned or when failing
+// to start or/and commit the transaction.
+func (db *DB) RunInTx(ctx context.Context, fn func(context.Context, *sqlx.Tx) error) error {
+	tx, err := db.BeginTxx(ctx, nil)
+	if err != nil {
+		return errors.Wrap(err, "DB.RunInTx: cannot start a database transaction")
+	}
+	// We don't expect meaningful errors from rolling back the tx other than the sql.ErrTxDone, so just ignore it.
+	defer func() { _ = tx.Rollback() }()
+
+	if err := fn(ctx, tx); err != nil {
+		return err
```

Comment:
No wrapping here on purpose?

Reply:
What should I wrap it with? Like it states in the method doc block, if the callback returns an error, then it is returned as-is. If in doubt, that provided callback should returned wrapped errors to clearly indicate what's going wrong.

Reply:
With a stacktrace.

Reply:
> What should I wrap it with?

With a stacktrace.

Reply:
Just wrapping with `errors.WithStack()` without additional messages? I can do this, but I just wanted to avoid another layer that meaninglessly wraps the error, which is not helpful for debugging and always expect the callback to return wrapped errors *with context specific messages*.

Reply:
Fixed!

Reply:
We had a conversation about this, and I wanted to add `errors.WithStack()` because otherwise there is no clear indication that the (error from the) callback actually went through this function.